lib/core/include/irods/library_features.h

```diff
@@ -28,4 +32,11 @@
 /// \since 4.3.1
 #define IRODS_HAS_LIBRARY_SYSTEM_ERROR        202301L
 
+/// Defined if the C++ client connection libraries support proxy users.
+///
+/// \since 4.3.1
+#define IRODS_HAS_PROXY_USER_SUPPORT_FOR_CLIENT_CONNECTION_LIBRARIES    202306L
```

Comment:
Perhaps this should be renamed to one of the following ...
```cpp
#define IRODS_HAS_ENHANCEMENT_PROXY_USERS_FOR_CONNECTION_LIBRARIES    202306L // A
#define IRODS_HAS_STRONG_TYPES_FOR_CPP_CONNECTION_LIBRARIES           202306L // B
```

Or something entirely different.

Currently, these macros are defined using the following convention ...
```cpp
#define IRODS_HAS_<type_of_thing_or_category>_<good_short_description>    YYYYMML
```

Reply:
A great naming discush.

Reply:
After discussion, we've decided to go with the following.
```cpp
IRODS_HAS_FEATURE_<description>
```

Reply:
Done.